cutting-a-release.md

```diff
@@ -8,28 +8,39 @@ Steps:
-- [ ] Once the docs look good, commit the changes.
-- [ ] Push the doc changes to master.
 - [ ] We use [Semantic Versioning](https://semver.org/) for all of our libraries. Decide which bump applies (major, minor, or patch) to the changes that will be relased. Here's a quick cheat sheet:
   - Use a Major version bump when you make incompatible API changes
   - Use a Minor version bump when you add functionality in a backwards-compatible manner
   - Use a Patch version bump when you make backwards-compatible bug fixes
-- [ ] Most repos have a section in the README that describes how to use the `make release` script. It looks something like [this](https://github.com/ethereum/web3.py#release-setup). Follow those instructions.
-  - If something goes wrong in the middle, you'll need to: 
-    - Figure out what went wrong and fix that problem
+- [ ] Make release notes before bumping the version number:
+  - Check the Makefile to see if there is a `make notes` command
+  - If there is a `make notes` command:
+    - [ ] From the command line run: `$ make notes bump=$$version part to bump$$`
+      - the "version part to bump" will usually be `major`, `minor`, or `patch`, but will sometimes be a devnum. Refer to the contributing guide in the docs for the library you're releasing to look at the syntax for
+      a devnum bump. For an example, check out the [web3.py Contributing Docs](https://web3py.readthedocs.io/en/stable/contributing.html#which-version-part-to-bump)
+    - [ ] Run `$ make docs` to view the release notes locally. Make sure everything looks like it should.
+  - If there is not a `make notes` command, you'll have to write the release notes manually.
+    - [ ] Look at the commits between the last version bump and the current one. Following the template already laid out in `docs/releases.rst`, write out what each unit of work does.
+      - Here are some categories that the units of work might fall under:
+        - Breaking Changes (only to be added in a major version bump)
+        - Bugfixes
+        - Deprecation
+        - Docs
+        - Features
+        - Internal
+        - Misc
+        - Performance
+        - Removal (only to be added in a major version bump)
+    - [ ] Run `$ make docs` to view the release notes locally. Make sure everything looks like it should.
+    - [ ] Once the docs look good, commit the changes.
+    - [ ] Push the doc changes to master.
+- [ ] Most repos have a section in either the README or the Contributing section of the docs that describes how to use the `make release` script. It looks something like [this](https://web3py.readthedocs.io/en/stable/contributing.html#releasing). Follow those instructions. Usually, it looks something like `$ make release bump=$$version part to bump$$`.
+  - Troubleshooting:
+    - If something goes wrong in the middle, you'll need to figure out what went wrong and fix that problem
```

Comment:
In our Makefiles we run a comand `git status -s -b | head -1 | grep "\.\.upstream/main"` that supposedly checks whether the local main branch is tracking the upstream/main branch. In my setup I have my main branch tracking my forked repo's main branch, not the upstream. That's why this command does not work for me. If there is something I can do with my setup to make this work, I am happy to do so.

Otherwise, we might consider using a different command to check that the upstream repo is correct. Seems like we could check for the upstream remote by just using `git remote`

```suggestion
    - If something goes wrong in the middle, you'll need to figure out what went wrong and fix that problem
    - If the problem persists, you may be able to skip the command and run the subsequent commands manually
```

Reply:
I'm glad you brought this up actually... I'm in the same boat and I usually just run through the steps manually (I prefer to anyways just to have more control). But I thought I was the only one and didn't mind if I kept doing it.

+1 to changing that step

Reply:
@reedsa You should be able to set tracking of upstream master (or main) by running:
`git branch -u upstream/master` or `git branch -u upstream/main`

But I like this addition, and will add!